### PR TITLE
Store collection in datacache

### PR DESCRIPTION
**Pull request by CasperWA**
# Description:

Store the session DLite Collection in the OTEAPI DataCache.

This is definitely not optimal, but it will ensure the DLite Collection "persists" between strategies - even when called in different sub-processes.

It should be considered a temporary "fix" to a general storage issue, which should be solved by DLite through a global state or service to govern DLite instances.

## Type of change:

- [ ] Bug fix.
- [x] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand, including clearly named variables?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: oteapi_dlite/strategies/function.py [hunk 1]
```diff
@@ -67,6 +66,13 @@ class DLiteStorageConfig(AttrDict):
         None,
         description="Configuration options for the local data cache.",
     )
+    global_configuration_additions: DLiteGlobalConfiguration = Field(
```
Comment (jesper-friis) [inline oteapi_dlite/strategies/function.py:69]: I think this should be considered as a quick-and-dirty solution as the strategy configurations should document the data resources or data sinks. A proper solution for pipeline-wise configurations should be sorted out in in issues https://github.com/EMMC-ASBL/oteapi-core/issues/211, 
https://github.com/EMMC-ASBL/oteapi-core/issues/177 and https://github.com/EMMC-ASBL/oteapi-core/issues/176.

Comment (CasperWA) [reply oteapi_dlite/strategies/function.py:69]: Or perhaps in an `oteapi-services` issue, really.

**Comment by CasperWA** [comment]
This PR blocks H2020-OpenModel/WP3-WrapperSDK#76.

File: tests/strategies/test_mapping.py [hunk 2]
```diff
@@ -1,57 +1,71 @@
...
-session2.update(mapper2.initialize(session2))
-session2.update(mapper2.get(session2))
-
-relations2 = set(coll2.get_relations())
-assert len(list(coll2.get_relations())) == len(relations2)
-assert (FORCES.forces, MAP.mapsTo, EMMO.Force) in relations2
-assert (ENERGY.energy, MAP.mapsTo, EMMO.PotentialEnergy) in relations2
+
+
+def test_mapping_without_prefixes() -> None:
```
Comment (jesper-friis) [inline tests/strategies/test_mapping.py:4]: I don't see that the changes in this file creates any value - it just makes the tests more obscure and makes it impossible, or at least much less convenient, to use the tests for debugging.

File: tests/utils/test_get_instance.py [hunk 3]
```diff
@@ -1,71 +1,81 @@
...
+    from pathlib import Path
 
-from oteapi_dlite.utils import get_instance
 
-# Setup
-thisdir = Path(__file__).absolute().parent
-entitydir = thisdir.parent / "entities"
-dlite.storage_path.append(f"{entitydir}/*.json")
+def test_instantiate_calcresults(entities_path: "Path") -> None:
+    """Test utils.get_instance().
```
Comment (jesper-friis) [inline tests/utils/test_get_instance.py:10]: Same comment as for the other test. The documentation is good and reduces the obfuscation of what is going on.

Still, I prefer the script.

File: oteapi_dlite/strategies/function.py [hunk 4]
```diff
@@ -113,6 +119,25 @@ def get(
         config = self.function_config.configuration
         cacheconfig = config.datacache_config
 
+        for addition in config.global_configuration_additions.storage_path:
```
Comment (jesper-friis) [inline oteapi_dlite/strategies/function.py:122]: This should be fractured out to a utility function

File: oteapi_dlite/strategies/mapping.py [hunk 5]
```diff
@@ -41,12 +54,32 @@ class DLiteMappingStrategy:
...
+        ) in config.global_configuration_additions.mapping_plugin_path:
+            dlite.mapping_plugin_path.append(addition)
+        for (
+            addition
+        ) in config.global_configuration_additions.python_storage_plugin_path:
+            dlite.python_storage_plugin_path.append(addition)
+        for (
+            addition
+        ) in config.global_configuration_additions.python_mapping_plugin_path:
+            dlite.python_mapping_plugin_path.append(addition)
```
Comment (jesper-friis) [inline oteapi_dlite/strategies/mapping.py:82]: Same as above, can be fractured out into an utility function.

File: oteapi_dlite/strategies/parse.py [hunk 6]
```diff
@@ -101,6 +111,25 @@ def get(
...
+        ) in config.global_configuration_additions.mapping_plugin_path:
+            dlite.mapping_plugin_path.append(addition)
+        for (
+            addition
+        ) in config.global_configuration_additions.python_storage_plugin_path:
+            dlite.python_storage_plugin_path.append(addition)
+        for (
+            addition
+        ) in config.global_configuration_additions.python_mapping_plugin_path:
+            dlite.python_mapping_plugin_path.append(addition)
```
Comment (jesper-friis) [inline oteapi_dlite/strategies/parse.py:131]: Same as above, can be fractured out into an utility function.

**Review by jesper-friis — CHANGES_REQUESTED** [review]

File: tests/strategies/test_mapping.py (hunk 2 above)
Comment (CasperWA) [reply tests/strategies/test_mapping.py:4]: We have agreed to follow the best practices of writing unit tests for our collective code within the team. This change follows that directive.

File: tests/utils/test_get_instance.py (hunk 3 above)
Comment (CasperWA) [reply tests/utils/test_get_instance.py:10]: We have agreed to follow the best practices of writing unit tests for our collective code within the team. This change follows that directive.

File: oteapi_dlite/strategies/function.py (hunk 4 above)
Comment (CasperWA) [reply oteapi_dlite/strategies/function.py:122]: Agreed. Also, I originally used `setattr()`, which would make this much more elegant, but as I couldn't get it to work at all, I couldn't check whether it was OK or not.

I think in the great picture of things, I will simply remove this functionality and configuration option altogether - what do you think?

File: oteapi_dlite/strategies/mapping.py (hunk 5 above)
Comment (CasperWA) [reply oteapi_dlite/strategies/mapping.py:82]: Will keep the discussion to the previous comment then, resolving this conversation.

File: oteapi_dlite/strategies/parse.py (hunk 6 above)
Comment (CasperWA) [reply oteapi_dlite/strategies/parse.py:131]: Will keep the discussion to the previous comment then, resolving this conversation.

File: oteapi_dlite/strategies/function.py (hunk 4 above)
Comment (CasperWA) [reply oteapi_dlite/strategies/function.py]: I have removed it altogether now.

**Review by jesper-friis — APPROVED** [review]

**Pull request merged**